internal/services/appconfiguration/app_configuration_key_resource.go

```diff
@@ -0,0 +1,358 @@
+package appconfiguration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration/sdk/1.0/appconfiguration"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type KeyResource struct {
+}
+
+var _ sdk.ResourceWithCustomizeDiff = KeyResource{}
+
+const (
+	KeyTypeVault        = "vault"
+	KeyTypeKV           = "kv"
```

Comment:
Minor niggle, given that KV can also mean KeyVault perhaps rename this const to `KeyTypeKeyValue`?

Reply:
Notice for folks reading this: Discussed it offline with Tom and decided to leave it as is.